### PR TITLE
Fix accessibilityState to support states not being set

### LINKED ISSUE #11791: UIA showing Expand/Collapsed property option on non-expandable components
### Problem Description

When focusing on non-expandable components (E.g Buttons, TouchableHighlight), the UIA tree shows the ExpandCollapsePattern as an option, when it should not be shown when focused on non expandable components.

Example:
image

As can be seen, the ExpandCollapsePattern exists when focusing on a Button in RNTester.

### Steps To Reproduce

1. Run Playground and open RNTester
2. Navigate to the Button page in the Accessibility section of RNTester
3. Open Accessibility Insights and focus on any of the buttons.
4. Look at UIA Tree output.

### Expected Results

Ideally there should be no ExpandCollapsePattern in the UIA Tree for Buttons, TouchableHighlights and other non expandable or collapsible elements. 

### CLI version

12.0.0-alpha.3

### Environment

```markdown
System:
  OS: Windows 10 10.0.22621
  CPU: "(24) x64 AMD Ryzen Threadripper PRO 3945WX 12-Cores     "
  Memory: 45.90 GB / 63.86 GB
Binaries:
  Node:
    version: 18.12.1
    path: C:\Program Files\nodejs\node.EXE
  Yarn:
    version: 1.22.19
    path: C:\Program Files (x86)\Yarn\bin\yarn.CMD
  npm:
    version: 8.19.2
    path: C:\Program Files\nodejs\npm.CMD
  Watchman: Not Found
SDKs:
  Android SDK: Not Found
  Windows SDK:
    AllowDevelopmentWithoutDevLicense: Enabled
    AllowAllTrustedApps: Enabled
    Versions:
      - 10.0.18362.0
      - 10.0.19041.0
      - 10.0.22000.0
      - 10.0.22621.0
IDEs:
  Android Studio: AI-213.7172.25.2113.9123335
  Visual Studio:
    - 17.6.33801.468 (Visual Studio Enterprise 2022)
    - 16.11.33529.622 (Visual Studio Enterprise 2019)
Languages:
  Java:
    version: 18.0.2.1
    path: C:\Program Files\Common Files\Oracle\Java\javapath\javac.EXE
  Ruby: Not Found
npmPackages:
  "@react-native-community/cli": Not Found
  react: Not Found
  react-native: Not Found
  react-native-windows: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: Not found
  newArchEnabled: Not found
iOS:
  hermesEnabled: Not found
  newArchEnabled: Not found
```

### Target Platform Version

None

### Target Device(s)

Desktop

### Visual Studio Version

Visual Studio 2022

### Build Configuration

Debug

### Snack, code example, screenshot, or link to a repository

_No response_

### PR DESCRIPTION
**Pull request by jonthysell**
## Description

Our existing implementation for `accessibilityState` does not support that the values of a state can be `null`. This means *every* View has *every* unset state with the default `false` value.

The problem stems from implementing the short-lived `accessibilityStates` prop at the same time as `accessibilityState`, but not removing the deprecated API right away.

This PR fixes our implementation to only include the states when explicitly set, and to otherwise clear them. This PR also removed the no longer required "opposite" states like `Collapsed` and `Unchecked`.

Resolves #11791

### Type of Change
- Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Why
Properly implement accessibility in preparation for doing the work correctly for Fabric.

Resolves #11791

### What

Removed the old unused states and updated FrameworkElementViewManager to allow for unset values rather than forcing the default false. Updated AccessbilityStateChecked to accept an enum of Unchecked, Mixed, or Checked. Updated the DynamicAutomationPeer to only express a View has the appropriate provider iff the matching state is present on the view.

## Screenshots
N/A

## Testing

Updated E2E test snapshots to include the AccessibilityRole and AccessibilityState properties in the XAML tree dumps.

 ###### Microsoft Reviewers: codeflow:open?pullrequest=https://github.com/microsoft/react-native-windows/pull/11792&drop=dogfoodAlpha

### CODE REVIEW

**Comment by jonthysell** [comment]
Need to get XamlTreeDump updated to test this properly, see https://github.com/asklar/XamlTreeDump/pull/10

**Comment by jonthysell** [comment]
Tagging as a breaking change since I am changing idls.

File: vnext/Microsoft.ReactNative/Views/DynamicAutomationPeer.cpp [hunk 1]
```diff
@@ -195,7 +198,12 @@ winrt::com_array<winrt::IRawElementProviderSimple> DynamicAutomationPeer::GetSel
 // ISelectionItemProvider
 
 bool DynamicAutomationPeer::IsSelected() const {
-  return GetAccessibilityState(winrt::Microsoft::ReactNative::AccessibilityStates::Selected);
+  bool selected = false;
+  try {
+    return DynamicAutomationProperties::GetAccessibilityStateSelected(Owner());
```
Comment (jonthysell) [inline vnext/Microsoft.ReactNative/Views/DynamicAutomationPeer.cpp:203]:
```suggestion
    selected = DynamicAutomationProperties::GetAccessibilityStateSelected(Owner());
```

File: vnext/Microsoft.ReactNative/Views/DynamicAutomationPeer.cpp [hunk 2]
```diff
@@ -195,7 +198,12 @@ winrt::com_array<winrt::IRawElementProviderSimple> DynamicAutomationPeer::GetSel
...
+  } catch (...) {
+  }
+  return false;
```
Comment (jonthysell) [inline vnext/Microsoft.ReactNative/Views/DynamicAutomationPeer.cpp:206]:
```suggestion
  return selected;
```

**Review by chiaramooney — APPROVED** [review]
Looks good to me

**Review by chiaramooney — COMMENTED** [review]
👍

**Pull request merged**